src/aliased_buffer.h

```diff
@@ -0,0 +1,193 @@
+
+#ifndef SRC_ALIASED_BUFFER_H_
+#define SRC_ALIASED_BUFFER_H_
+
+#include "v8.h"
+
+namespace node {
+
+/**
+ * This class encapsulates the technique of having a native buffer mapped to
+ * a JS object. Writes to the native buffer can happen efficiently without
+ * going through JS, and the data is then available to user's via the exposed
+ * JS object.
+ *
+ * While this technique is computationaly efficient, it is effectively a
+ * write to JS program state w/out going through the standard
+ * (monitored) API. Thus any VM capabilities to detect the modification are
+ * circumvented.
+ *
+ * The encapsulation herein provides a placeholder where such writes can be
+ * observed. Any notification APIs will be left as a future exercise.
+ */
+template <class NativeT, class V8T>
+class AliasedBuffer {
+ public:
+  AliasedBuffer(v8::Isolate* isolate, const size_t count) :
+    isolate_(isolate),
+    count_(count),
+    byte_offset_(0),
+    freeBuffer_(true) {
+    const v8::HandleScope handle_scope(this->isolate_);
+
+    const int sizeInBytes = sizeof(NativeT) * count;
+
+    // allocate native buffer
+    this->buffer_ = reinterpret_cast<NativeT*>(
+      calloc(count, sizeof(NativeT)));
```

Comment:
Two problems here. 1) please use `node::UncheckedCalloc()` instead of using `calloc()` directly. 2) You're not checking if `this->buffer_ == nullptr` afterwards.

Reply:
Thanks @trevnorris, will do.  Is there an example of fail-fast behavior when out of memory occurs?  e.g., is there a specific error code to return?

Reply:
Good catch. I tend to completely forget that we have those Calloc wrappers there.